### ISSUE TITLE
ModuleDefinition.Assembly is null in multi-module assemblies

### ISSUE THREAD

**Issue by sajagi** [jbevain/cecil#652]
I am not sure whether this qualifies as a bug or is it intentional. `ModuleDefinition.Assembly` property in `AssemblyDefinition.Modules` is `null` for modules which are not defined in the same dll as the assembly. Given that the the assembly definition always creates new instances of module definitions it would make sense to keep the link back to the assembly. 

```csharp
public void Test()
{
    var module = ModuleDefinition.ReadModule(@"c:\Windows\Microsoft.NET\Framework64\v4.0.30319\System.EnterpriseServices.dll");
    
    foreach (var m in module.Assembly.Modules)
        Console.WriteLine($"{m.Name} assembly: {m.Assembly?.Name.Name ?? "null"}");

    // prints: 
    // System.EnterpriseServices.dll assembly: System.EnterpriseServices
    // System.EnterpriseServices.Wrapper.dll assembly: null
}
```

The problem with current situation is that currently there is no way to associate TypeDefinitions in such auxiliary modules back to the assembly they were defined in. 

Thanks for your hard work!

**Comment by jbevain** [comment]
I think the original intent was that there's no Assembly row in a multi-module assembly.
So in theory, a multi module assembly could even belong to 2 assemblies.
Also I like to pretend that multi-module don't really exist :)

**Comment by jbevain** [comment]
I think the way to go would be:

If you use `AssemblyDefinition.ReadAssembly` on a multi-module assembly, the `.Assembly` property should return the main assembly.
If you use `ModuleDefinition.ReadModule` on a `.netmodule`, the `.Assembly` property should return null.

**Comment by jbevain** [comment]
Fixed by #730
